src/common/bitstream.h

```diff
@@ -301,13 +301,31 @@ class BitStream {
         }
     }
 
-    void read(std::string& str) {
+    /**
+     * Read the length of the following string from the bitstream as a big-endian number.
+     *
+     * PlanetSide packets mainly use length-prefaced strings, where the length is either one byte or two bytes.
+     * If the length is less than 128, the first bit of the field is "on" and the length is the remaining seven bits.
+     * If the length is 128 or greater, the first bit of the field is "off" and the length is the remaining fifteen bits.
+     * In other words: 0 <= n < 128 is 1 byte, from 0x80 to 0xFF;
+     * and, 128 <= n < 32767 is 2 bytes, from 0x0080 to 0x7FFF.
+     *
+     * @param bitStream the stream, with the current position at the length field
+     * @return the length
+     */
+    uint16_t readStringLength() {
         uint16_t strLen = 0;
         if (readBit()) {
             readBits((uint8_t*)&strLen, 7);
-        } else {
+        }
+        else {
             readBits((uint8_t*)&strLen, 15);
         }
+        return strLen;
+    }
+
+    void read(std::string& str) {
+		uint16_t strLen = readStringLength();
```

Comment:
I know we haven't come to a conclusion on the whole tabs vs spaces thing in the style guide yet. For now though, I think we should stay consistent with the 4-spaces so that we can convert it all at once if we decide to go to tabs.